sessionctx/variable/tidb_vars.go

```diff
@@ -1195,7 +1195,7 @@ const (
 	DefTiDBEnableFastReorg                         = true
 	DefTiDBDDLDiskQuota                            = 100 * 1024 * 1024 * 1024 // 100GB
 	DefExecutorConcurrency                         = 5
-	DefTiDBEnableNonPreparedPlanCache              = false
+	DefTiDBEnableNonPreparedPlanCache              = true
```

Comment:
@qw4990 you sure wanna enable it by default? 
<img width="139" alt="image" src="https://user-images.githubusercontent.com/3692139/232721964-35808954-bdb1-4305-a85c-ff5b70accbe2.png">

Reply:
Yeah, why not... 1) I've finished all tests, 2) we can pass all existing test cases with this feature enabled, 3) we can get some significant performance benefits on TP workloads from this feature (10+%).
So is there any reason that can keep us from enabling this feature by default... From R&D's perspective, we have done all we can to test it (add more UTs, let it can pass all existing cases, utilize some testing tools to test it...)